File: uchile_navigation_utils/src/CostmapCleaner.cpp

#include "ros/ros.h"
#include <std_srvs/Empty.h>
#include <uchile_util/ParameterServerWrapper.h>

int main(int argc, char **argv) {

	ros::init(argc, argv, "costmap_cleaner");
	ros::NodeHandle priv("~");

	float dt; // [s]
	uchile_util::ParameterServerWrapper psw;
	psw.getParameter("dt", dt, 5);

	ros::ServiceClient clear_client = priv.serviceClient<std_srvs::Empty>("/bender/nav/move_base/clear_costmaps");
	while ( ros::ok() && !clear_client.waitForExistence(ros::Duration(3.0)) ) ;

	ROS_INFO("Ready to clean ...");

	ros::Rate r(1.0/dt);
	std_srvs::Empty srv;
	while(ros::ok())
	{
		try {
			clear_client.call(srv);
		} catch (std::exception &e) {
			ROS_WARN_STREAM(e.what());
		}
		r.sleep();
	}
	printf("\nQuitting... \n\n");
	return 0;
}

File: uchile_navigation_utils/src/costmap_cleaner.py
#!/usr/bin/env python
import rospy
from std_srvs.srv import Empty


if __name__ == "__main__":
	rospy.init_node('costmap_cleaner')
	service_name = rospy.get_param("service_name", "/maqui/nav/move_base/clear_costmaps")   
	dt = rospy.get_param('dt', 5.0)
	rospy.loginfo("Waiting for service: {}".format(service_name))
	rospy.wait_for_service(service_name)
	print "Service {} available".format(service_name)
	service_client = rospy.ServiceProxy(service_name,Empty)

	rate = rospy.Rate(1.0/dt)
	while(not rospy.is_shutdown()):
		try:
			service_client()
			rospy.loginfo("Calling service: {}".format(service_name))
		except rospy.ServiceException, e:
			rospy.logwarn("Service call failed: {}".format(e))

		rate.sleep()
	rospy.loginfo("Exit...")
	